scanner/scanner.go

```diff
@@ -269,11 +269,48 @@ func humanTime(seconds int) string {
 	return s
 }
 
+func (s *Scanner) logThroughput(treeSize int64, stop <-chan bool) {
+	const wndSize = 15
+	wnd := make([]int64, wndSize)
+	wndTotal := int64(0)
+
+	ticker := time.NewTicker(time.Second)
+	for slot, slots, prevCnt := 0, 0, int64(0); ; slot = (slot + 1) % wndSize {
+		select {
+		case <-stop:
```

Comment:
Perhaps a better approach would be to drop the `stop` channel and just close the `ticker` from the outside.

Reply:
Apparently `ticker.Stop()` doesn't close `ticker.C`, and there is no way to close that channel manually. The way to terminate here is indeed having an additional channel (see Example [here](https://golang.org/pkg/time/#NewTicker)).